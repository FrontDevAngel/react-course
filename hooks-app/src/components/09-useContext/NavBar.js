import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className='navbar navbar-expand-sm navbar-dark bg-dark px-3'>
            <Link className='navbar-brand' to='./'>
                UseContext
            </Link>
            <div className='navbar-nav'>
                <NavLink
                    activeClassName='active'
                    className='nav-item nav-link'
                    exact
                    to='./'>
                    Home
                </NavLink>
                <NavLink
                    activeClassName='active'
                    className='nav-item nav-link'
                    exact
                    to='./about'>
                    About
                </NavLink>
                <NavLink
                    activeClassName='active'
                    className='nav-item nav-link'
                    exact
                    to='./login'>
                    Login
                </NavLink>
            </div>
        </nav>
    );
};
