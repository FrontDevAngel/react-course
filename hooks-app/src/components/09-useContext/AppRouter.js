import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
} from 'react-router-dom';
import { NavBar } from './NavBar';

import { HomeScreen } from './HomeScreen';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className='container'>
                    <Switch>
                        <Route exact path='/' component={HomeScreen} />
                        <Route exact path='/about' component={AboutScreen} />
                        <Route exact path='/login' component={LoginScreen} />

                        {/* Redirecci√≥n si no cumple con ninguna de las anteriores */}

                        {/* Dirigir al componente ////////////////////// */}
                        {/* <Route component={ FailScreen } /> */}

                        {/* Dirigir al componente existente ////////////////////// */}
                        {/* <Route component={ HomeScreen } /> */}

                        {/* Redirecta a un Path ////////////////////// */}
                        <Redirect to='/' />
                    </Switch>
                </div>
            </div>
        </Router>
    );
};
