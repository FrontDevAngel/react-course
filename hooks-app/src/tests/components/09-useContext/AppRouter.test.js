import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { userDemo } from '../fixtures/userDemo';

/**
 * No debemos de probar router de momento porque la gente de router ya lo probo y
 * no hemos pasado argumentos importantes en el router para necesitar identificarlos
 */

describe('Pruebas en el <AppRouter/>', () => {
    const wrapper = mount(
        <UserContext.Provider value={{ user: userDemo }}>
            <AppRouter />
        </UserContext.Provider>,
    );

    it('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
