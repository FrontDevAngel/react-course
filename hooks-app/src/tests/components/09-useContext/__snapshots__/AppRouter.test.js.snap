// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <AppRouter/> Debe mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark px-3"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      UseContext
    </a>
    <div
      className="navbar-nav"
    >
      <a
        aria-current="page"
        className="nav-item nav-link active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "id": 123456,
   "name": "√Ångel Pantoja"
}
      </pre>
    </div>
  </div>
</div>
`;
